cli/azd/pkg/project/framework_service_dotnet.go

```diff
@@ -181,3 +200,19 @@ func normalizeDotNetSecret(key string) string {
 	// ":".
 	return strings.ReplaceAll(key, "__", ":")
 }
+
+func findProjectFile(path string, dotnetProjectFile string) ([]string, error) {
+	files, err := filepath.Glob(path + "/*.*proj")
+	if err != nil {
+		return files, fmt.Errorf("error: checking project file in %s: %w", path, err)
+	}
+	if len(files) == 0 {
+		return files, fmt.Errorf("no project file (.csproj or .vbproj or .fsproj) found")
+	} else if len(files) > 1 && dotnetProjectFile == "" {
+		return files, fmt.Errorf("there are multiple project files in %s. "+
+			"Please add the project file path with row 'dotnetProjectFile' in azure.yaml", path)
+	} else if len(files) > 1 && dotnetProjectFile != "" {
+		files = []string{dotnetProjectFile}
+	}
+	return files, err
```

Comment:
You don't need to check `len(files)` twice.
And you can return the project file directly, instead of an array (the array is only ever returning one value in it anyway)

```suggestion
        filesFound :+ len(files)
	if filesFound  == 0 {
		return "", fmt.Errorf("no project file (.csproj or .vbproj or .fsproj) found")
	}
        if filesFound  == 1 {
		return files[0], err
	}
	// At this point, you know filesFound is greater than 1
	if dotnetProjectFile == "" {
    	     return files, fmt.Errorf("there are multiple project files in %s. "+
			"Please add the project file path with row 'dotnetProjectFile' in azure.yaml", path)
	}
	// Need to check if the dotnetProjectFile is in the list of projects
	for _, foundProject := range files {
	     if foundProject.Name == dotnetProjectFile {
	       return dotnetProjectFile, nil
	     }
	}
	return "", fmt.Errorf("expecting to find project %s, but it was not found", dotnetProjectFile)
	
```